tools/codegen/src/ast/raw/mod.rs

```diff
@@ -1,16 +1,43 @@
 use std::path::PathBuf;
 
+#[cfg(feature = "compiler-plugin")]
+use serde::{Deserialize, Serialize};
+
 use property::Property;
 
 mod utils;
 
 #[derive(Debug, Default, Property)]
 pub(crate) struct Ast {
+    syntax_version: Option<SyntaxVersion>,
     namespace: String,
     imports: Vec<ImportStmt>,
     decls: Vec<TopDecl>,
 }
 
+impl Default for SyntaxVersion {
+    fn default() -> Self {
+        Self { major: 1, minor: 0 }
+    }
+}
+
+#[cfg(not(feature = "compiler-plugin"))]
+#[derive(Debug, Clone, PartialEq, Eq, Property)]
+#[property(get(public))]
+pub struct SyntaxVersion {
+    major: usize,
+    minor: usize,
+}
+
+#[cfg(feature = "compiler-plugin")]
+#[derive(Debug, Clone, PartialEq, Eq, Property, Deserialize, Serialize)]
```

Comment:
This can be merged into a single structure

Reply:
> This can be merged into a single structure

Thank you, this can be merged into 
```rust
#[derive(Debug, Clone, PartialEq, Eq, Property)]
#[property(get(public))]
#[cfg_attr(
    feature = "compiler-plugin",
    derive(Deserialize, Serialize),
    serde(deny_unknown_fields)
)]
pub struct SyntaxVersion {
    major: usize,
    minor: usize,
}

```